viper.go

```diff
@@ -2034,6 +2038,24 @@ func (v *Viper) AllKeys() []string {
 	return a
 }
 
+// bindStruct binds a struct to the environment by extracting all Viper keys
+// from it and binding each to a ENV variable using BindEnv.
+func (v *Viper) bindStruct(input interface{}) error {
+	envKeysMap := map[string]interface{}{}
+	if err := mapstructure.Decode(input, &envKeysMap); err != nil {
+		return err
+	}
+
+	structKeys := v.flattenAndMergeMap(map[string]bool{}, envKeysMap, "")
```

Comment:
These keys should be merged with the list of AllKeys instead. No need to call BindEnv.